### DISCUSSION TITLE
Best way to create a grid of characters?

### DISCUSSION THREAD

**Discussion by SymmetricChaos** [emilk/egui#1312] [Q&A]
I'm making an old fashioned cipher suite and to display information I need to draw grids of evenly spaced characters. However label and TextEdit seem to take up specifically rectangular spaces and everything is stretched out even if item spacing it set to zero. Is there a way to get something like the grid in the FontBook from the demo but without using buttons?

What I've tried is roughly this

```
pub fn letter_grid(ui: &mut egui::Ui, rows: Vec<String>) {
    egui::Grid::new("letter_grid").show(ui, |ui| {
        ui.spacing_mut().item_spacing.x = 0.0;
        for row in rows {
            for c in row.chars() {
                ui.add_sized([0.0, 0.0], Label::new(&String::from(c)));
            }
            ui.end_row()
        }
    });
}
```

**Comment by emilk** [comment]
Adding zero-sized labels seems like a bad idea. You code should work if you just did `ui.label(c.to_string());` instead of `ui.add_sized(…)`.

 You can use `ui.put` to get full control.
